encoder_hexa_v_4.rb

```diff
@@ -1,26 +1,22 @@
-require 'CSV'
-$csv_array = CSV.open("data/char_list.csv").read()
-$input_array_index = 0
-$csv_array_y = 0
-$csv_array_x = 0
-$encoded = Array.new
-enc = gets.chomp.split("")
+require 'csv'
+encoding_array = CSV.open("data/char_list.csv").read()
 
-def letter (character)
-	case character
-		when $csv_array[$csv_array_y][$csv_array_x]
-			$encoded << $csv_array[$csv_array_y][$csv_array_x += 1]
-			$input_array_index += 1
-			$csv_array_y = 0
-			$csv_array_x = 0
-		else
-			$csv_array_y += 1
-	end
+letters_to_encode = gets.chomp.split("")
+
+def encode(letter, csv_array)
+  letter_and_encodings_array = csv_array.select { |l_a_e| letter == l_a_e[0] }

```

Comment:
Of course usually is the language of the review on Github English :slightly_smiling_face: